apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py

```diff
@@ -498,29 +506,40 @@ def render(element_html: str, data: pl.QuestionData) -> str:
     elif data["panel"] == "submission":
         if grading_method is GradingMethodType.EXTERNAL:
             return ""  # external grader is responsible for displaying results screen
+        feedback_type = pl.get_enum_attrib(
+            element, "feedback", FeedbackType, FEEDBACK_DEFAULT
+        )
+        show_first_wrong = feedback_type in [
+            FeedbackType.FIRST_WRONG,
+            FeedbackType.FIRST_WRONG_VERBOSE,
+        ]
+        student_submission = [
+            {
+                "inner_html": attempt["inner_html"],
+                "indent": (attempt["indent"] or 0) * TAB_SIZE_PX,
+                "badge_type": attempt["badge_type"] if show_first_wrong else "",
+                "icon": attempt["icon"] if show_first_wrong else "",
+                "first_wrong": show_first_wrong,
+                "distractor_feedback": attempt.get("distractor_feedback"),
+                "show_distractor_feedback": "distractor_feedback" in attempt
+                and attempt["badge_type"] == "badge-danger"
```

Comment:
Is the idea here to check whether the `attempt` was marked correct or not? I think there are more expressive ways of doing this than checking whether the badge is equal to `"badge-danger"`.

In general, it's best if we can reduce the number of comparisons involving string literals, as this makes the code harder to read and usually indicates something can be simplified.

Reply:
Its actually more subtle than this, its only if the block is the first one that was wrong. 

Anyway, I agree with you about string comparisons, it was just a hack while refactoring and I figured out a better way to do it.